wagtail/wagtailimages/views/images.py

```diff
@@ -228,9 +229,18 @@ def delete(request, image_id):
         return permission_denied(request)
 
     if request.method == 'POST':
-        image.delete()
-        messages.success(request, _("Image '{0}' deleted.").format(image.title))
-        return redirect('wagtailimages:index')
+        try:
+            image.delete()
+            messages.success(request, _("Image '{0}' deleted.").format(image.title))
+            return redirect('wagtailimages:index')
+        except ProtectedError as e:
+            item = e.protected_objects[0]
+            messages.warning(request, _(
+                "Image '{0}' can't be deleted because is in use in '{1}'."
+            ).format(image.title, item.title), buttons=[
```

Comment:
This will only work if the item is a Page. If the item was a snippet, or some other model without a `title` attribute, this would cause an error.